### PR TITLE
feat(corpus_candidate_sets): Add German candidates sets and refactor flows [TDP-223]

### PR DESCRIPTION
**Pull request by mmiermans**
## Goal
Generate German corpus candidate sets for:
- collections_by_recency.py ('Popular Collections')
- new_tab_not_syndicated_or_collection.py ('Recommended Reads')
- topic slates

## Implementation Decisions
- Generate candidate sets based on a config instead to get rid of duplicate flow code.
- Corrected `data` type of `PocketSnowflakeQuery` to `Union[tuple, Dict]`. A `dict` can be used for named query arguments.
 
## References

- [JIRA ticket](https://getpocket.atlassian.net/browse/TDP-223)
- [PRD](https://getpocket.atlassian.net/wiki/spaces/CP/pages/2809135186/PRD%3A+German+Unified+Home+V.1)

### CODE REVIEW

**Review by lfishhead — APPROVED** [review]
this looks good to me.  i made two comments, one about reviewing a query with G and another about how tightly to couple candidate set generation across locales.  happy to discuss as needed.

File: src/flows/recommendation_api/corpus_candidate_sets/corpus_candidate_sets.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+UNION ALL
+
+SELECT
+    german_curated_corpus_candidate_set_id as "CORPUS_CANDIDATE_SET_ID",
+    concat('de_de/', LOWER(corpus_topic_id)) as "NAME",
+    corpus_topic_id as "CORPUS_TOPIC_ID",
+    'NEW_TAB_DE_DE' as "SCHEDULED_SURFACE_ID"
+FROM analytics.dbt.static_corpus_candidate_set_topics
+"""
+
```
[the remark this answers is not in the document]
Comment (mmiermans) [reply src/flows/recommendation_api/corpus_candidate_sets/corpus_candidate_sets.py:41]: Agreed: I'm thinking that a human-readable / structured identifier. I'm working a bit towards that by adding a `name` parameter to configs.

File: src/flows/recommendation_api/corpus_candidate_sets/sql/life_hacks.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        topic as "TOPIC",
+        publisher as "PUBLISHER",
+        scheduled_corpus_item_scheduled_at as "REVIEW_TIME"
+    FROM "ANALYTICS"."DBT"."SCHEDULED_CORPUS_ITEMS"
+    WHERE SCHEDULED_CORPUS_ITEM_SCHEDULED_AT < current_timestamp()
+    AND CORPUS_ITEM_LOADED_FROM = 'MANUAL'  -- should this be removed?
+    AND SCHEDULED_SURFACE_ID = 'NEW_TAB_EN_US'
+    AND NOT is_syndicated
+    AND NOT is_collection
+)
```
[the remark this answers is not in the document]
Comment (mmiermans) [reply src/flows/recommendation_api/corpus_candidate_sets/sql/life_hacks.sql:26]: There's a cleanup ticket [TDP-209](https://getpocket.atlassian.net/browse/TDP-209) to simplify these queries based on work that G did in Dbt.

[the remark this answers is not in the document]
Comment (mmiermans) [reply src/flows/recommendation_api/corpus_candidate_sets/sql/life_hacks.sql:26]: G recently made changes to ensure `APPROVED_CORPUS_ITEMS` and `SCHEDULED_CORPUS_ITEMS` only contain items that are approved and scheduled now (as opposed to at any time in the past).

**Pull request merged**